kernel/src/engine/default/storage.rs

```diff
@@ -0,0 +1,25 @@
+use hdfs_native_object_store::HdfsObjectStore;
+use object_store::parse_url_opts as object_store_parse_url_opts;
+use object_store::path::Path;
+use object_store::{Error, ObjectStore};
+use url::Url;
+
+pub fn parse_url_opts<I, K, V>(url: &Url, options: I) -> Result<(Box<dyn ObjectStore>, Path), Error>
```

Comment:
could we add a test for this? for now it could just make sure that `hdfs`/`viewfs` work as expected. 

I'd like to expand it to testing that we do the expected thing for aws/azure/gcp as well, but that doesn't need to be part of this pr.

Reply:
I've added a simple integration test using the MiniDfs cluster from `hdfs-native`, the test is behind an `integration-test` feature flag.

You can run the test like so:

```bash
cargo test --features integration-test,cloud --test hdfs
```

Reply:
Awesome thanks! Could you add that command as a comment in the file? :)